torchtune/modules/attention.py

```diff
@@ -161,6 +192,10 @@ def forward(
         # input has shape [b, s, d]
         bsz, seq_len, _ = x.shape
 
+        if self.kv_cache and input_pos is None:
+            cache_size = self.kv_cache.size
+            input_pos = torch.arange(cache_size, cache_size + seq_len, device=x.device)
```

Comment:
This part feels a bit weird to me. So we are relying on KV cache to infer the number of tokens seen by the positional embeddings thus far during inference? Why not just define a variable in positional embeddings for this? And just as a sanity check: we never have to worry about sample packing because this is an inference-only path, right?

Reply:
We're relaying on kv_cache to track the input_pos during coding because, the amount cached and amount decoded are the same thing. For me the positional embeddings make less sense for this because they have nothing to do with input_pos, they are just converting positional information into embedding form.